knative-kubecon/README.md

```diff
@@ -1,3 +1,219 @@
+
+Knative provides a mechanism called **BuildTemplates**, where you define a blueprint for a build, which contains the
+arguments that that build might need to do its job of building the image in the end. An example of such a template can
+be seen below. This template allows building images through Knative Build based on OpenShift's Build capability, so you
+can use the tools you're used to while taking full advantage of Knative Build on top of it.
+
+```yaml
+apiVersion: build.knative.dev/v1alpha1
+kind: BuildTemplate
+metadata:
+  name: openshift-builds
+spec:
+  parameters:
+  - name: IMAGE
+    description: The name of the image to push
+  - name: NAME
+    description: Build configuration name
+  - name: IMAGE_STREAM
+    description: The image stream to use as input for the build
+  - name: TO_DOCKER
+    description: Push the image to a Docker repository or not (true by default)
+    default: "false"
+  - name: DIRECTORY
+    description: The directory containing the app
+    default: /workspace
+  - name: OC_BUILDER_IMAGE
+    description: The name of the builder image to use
+    default: docker.io/vdemeester/kobw-builder:0.0.1
+  steps:
+  - name: kobw-create-or-update
+    image: "${OC_BUILDER_IMAGE}"
+    args: ["create", "--name=${NAME}", "--image=${IMAGE}", "--image-stream=${IMAGE_STREAM}", "--to-docker=${TO_DOCKER}", "."]
+    workingDir: "${DIRECTORY}"
+    env:
+      - name: NAMESPACE
+        valueFrom:
+          fieldRef:
+            fieldPath: metadata.namespace
+  - name: kobw-run
+    image: "${OC_BUILDER_IMAGE}"
+    args: ["run", "--name=${NAME}"]
+    env:
+      - name: NAMESPACE
+        valueFrom:
+          fieldRef:
+            fieldPath: metadata.namespace
+```
+
+To achieve the desired effect, this template wraps an OpenShift Build entity to perform the build of the desired application.
+To "install" that template, we need to `oc apply` it by:
+
+```bash
+$ oc apply -f build/000-build-template.yaml
+```
+
+A **Build**, as seen below, then references such a BuildTemplate and provides it with the arguments needed to perform the build.
+
+```yaml
+apiVersion: build.knative.dev/v1alpha1
+kind: Build
+metadata:
+  name: oc-build-1
+spec:
+  source:
+    git:
+      url: https://github.com/openshift-cloud-functions/serving-function
+      revision: master
+  template:
+    name: openshift-builds
+    arguments:
+    - name: IMAGE_STREAM
+      value: golang:1.11
+    - name: IMAGE
+      value: "helloworld:latest"
+    - name: NAME
+      value: helloworld-build
+```
+
+In this particular case, the source code is taken from a git repository's master branch. The arguments to the template then define
+that we want to build a Golang based application and want to name the image `helloworld:latest`.
+
+Now we could go ahead and run this build on its own by applying it like the template above and using
+`oc apply -f build/010-build.yaml`, but it's much more interesting to see how it's stringed together
```

Comment:
move to separate line.  easy to overead this


also, I get 

```
NAME               READY     STATUS	  RESTARTS   AGE
oc-build-1-fd56b   0/1       Init:Error   0          2m




```

Reply:
not exaclty sure if steps from: https://github.com/openshift-cloud-functions/demos/blob/7678bc5001c44979c397276089fe562685c7f6fe/knative-kubecon/build/README.md#example 

are missing

Reply:
They are, I'm putting them in the general install.sh script for ease of use.